src/rewards/poprewards.cpp

```diff
@@ -0,0 +1,114 @@
+#include <cassert>
+#include <vector>
+#include <veriblock/entities/atv.hpp>
+#include <veriblock/rewards/poprewards.hpp>
+#include <veriblock/rewards/poprewards_calculator.hpp>
+
+namespace altintegration {
+
+static int getBestPublicationHeight(
+    const std::vector<VbkEndorsement>& endorsements,
+    const VbkBlockTree& vbk_tree) {
+  int bestPublication = -1;
+  for (const VbkEndorsement& e : endorsements) {
+    auto* block = vbk_tree.getBlockIndex(e.blockOfProof);
+    if (block == nullptr) continue;
+
+    if (block->height < bestPublication || bestPublication < 0)
+      bestPublication = block->height;
+  }
+  return bestPublication;
+}
+
+PopRewardsBigDecimal PopRewards::scoreFromEndorsements(
+    const AltBlock& endorsedBlock) {
+  PopRewardsBigDecimal totalScore = 0.0;
+
+  /// TODO: in Java we were limiting the search in containingBlocks list.
+  ///      Maybe we should do the same here as well.
+  auto endorsements = erepo_.get(endorsedBlock.getHash());
```

Comment:
these endorsements will be from ALL chains. here you need to filter them...

Reply:
`containingHash` must be on AltTree's current best chain, and `blockOfProof` must be on VbkTree's current best chain -> this is validity rule

Reply:
leave it as is, we will add new way of getting endorsements from current chain soon

Reply:
UPD: use `BlockIndex::containingEndorsements`